### PR TITLE
[JUJU-1130] Refactor `model-defaults` to use ConfigCommandBase

### PR DESCRIPTION
**Pull request by barrettj12**
Continues work done in #13983, #14014, #14036.

I made the following changes to the `model-defaults` CLI, to bring it in line with the other config commands:
- Yaml files must now be specified by the `--file` flag.
- You can't perform multiple actions at once (e.g. resetting and setting).

I've also introduced the `--cloud` flag which is required to specify the cloud/region. Arguments are accepted as follows:
```
    --cloud=<cloud>             (specified cloud, all regions)
    --cloud=<cloud>/<region>    (specified cloud and region)
    --cloud=/<region>           (default cloud, specified region)
```
This prevents ambiguity when clouds/regions/default keys have the same name, and significantly simplifies the argument parsing code.

Finally, I made some small changes to `ConfigCommandBase` in a separate commit:
- with a call like `config key1 key2 --reset key3,key4`, I've changed the error to something more useful:
  ~`cannot specify multiple keys to get`~
  `cannot use --reset flag and get value simultaneously`
- Simplified the code for the parsing tests.

### CODE REVIEW

**Comment by barrettj12** [comment]
In a future PR, we will change this so that setting/resetting simultaneously is allowed.

File: featuretests/cmd_juju_model_test.go [hunk 1]
```diff
@@ -221,7 +221,7 @@ func (s *cmdModelSuite) TestModelDefaultsResetRegion(c *gc.C) {
 	err := s.State.UpdateModelConfigDefaultValues(map[string]interface{}{"special": "known"}, nil, &environscloudspec.CloudRegionSpec{"dummy", "dummy-region"})
 	c.Assert(err, jc.ErrorIsNil)
 
-	s.run(c, "model-defaults", "dummy-region", "--reset", "special")
+	s.run(c, "model-defaults", "--cloud", "/dummy-region", "--reset", "special")
```
Comment (wallyworld) [inline featuretests/cmd_juju_model_test.go:224]: Is `/dummy-region` valid? I've only ever seen it fully specified as `cloud/region`. The latter is better as it's explicit.

**Review by wallyworld — COMMENTED** [review]
Quick look through, looks good (with a followup as discussed to allow set and reset).
However, allowing "/region" is not something I think we've done before. Check the other places where `--cloud` is used. We want to be consistent. I tend to think we'd want either `--cloud cloud` or `--cloud cloud/region`. Maybe `--cloud /region` is ok (I'd like another opinion) but either way, we need to ensure consistency.

**Comment by wallyworld** [comment]
Yeah, I realise model-defaults takes a region, it was questioning the new syntax used. We use `--cloud` elsewhere and this would be the first time (to my recollection) we'd be supporting the `/region` syntax.

**Comment by barrettj12** [comment]
Okay, I've changed the cloud/region syntax to match the `add-k8s` command. Now it looks like:
```
    --cloud=<cloud>                    (specified cloud, all regions)
    --region=<region>               (default cloud, specified region)
    --region=<cloud>/<region>            (specified cloud and region)
    --cloud=<cloud> --region=<region>    (specified cloud and region)
```

**Review by wallyworld — APPROVED** [review]
Thanks for updating cloud/region.
Let's land and do the next bit.

File: prinfo.json [hunk 2]
```diff
@@ -0,0 +1 @@
...:"Upgrade (latest/stable, microk8s)","status":"COMPLETED","conclusion":"FAILURE","startedAt":"2022-05-18T04:04:44Z","completedAt":"2022-05-18T04:07:59Z","detailsUrl":"https://github.com/juju/juju/runs/6482727881?check_suite_focus=true"},{"__typename":"CheckRun","name":"Upgrade (latest/beta, localhost)","status":"COMPLETED","conclusion":"SUCCESS","startedAt":"2022-05-18T04:04:44Z","completedAt":"2022-05-18T04:04:50Z","detailsUrl":"https://github.com/juju/juju/runs/6482727928?check_suite_focus=true"},{"__typename":"CheckRun","name":"Upgrade (latest/beta, microk8s)","status":"COMPLETED","conclusion":"SUCCESS","startedAt":"2022-05-18T04:04:44Z","completedAt":"2022-05-18T04:04:49Z","detailsUrl":"https://github.com/juju/juju/runs/6482727971?check_suite_focus=true"},{"__typename":"StatusContext","name":"","context":"check-multi-juju","state":"SUCCESS","status":"","conclusion":"","startedAt":"0001-01-01T00:00:00Z","completedAt":"0001-01-01T00:00:00Z","detailsUrl":"","targetUrl":"https://juju-temp-jenkins.haza.dev/job/github-juju-check-jobs/334/"}],"title":"[JUJU-1057] Fix GitHub Actions on `develop` branch","updatedAt":"2022-05-19T01:09:45Z","url":"https://github.com/juju/juju/pull/14006"}
```
Comment (manadart) [inline prinfo.json:1]: Was this supposed to be committed?

Comment (barrettj12) [reply prinfo.json:1]: Oh dear, don't know how that snuck in there. I'll delete it

**Pull request merged**